os/kernel/init/os_start.c

```diff
@@ -599,6 +600,13 @@ void os_start(void)
 
 		/* Perform any processor-specific idle state operations */
 
-		up_idle();
+#ifdef CONFIG_PM
+		pm_idle();
+#else
+		/* Sleep until an interrupt occurs to save power */
+#if 0
+		asm("WFI");			/* For example */
```

Comment:
wfi should not be called from kernel. wfi is arm specific, hence should not be called from common code.
I would advise to provide an API for using cpu specific instruction for ex: up_cpuidle(), where each architecture would implement this with generic call. Incase of arm it would be asm("wfi"); for other arch it will be other arch specific instruction.  It should be enabled always, hence you can remove it from #else block.

Ex(for ARM):
File : os/arch/arm/src/common/up_cpuidle.c 
Func : 
void up_cpuidle(void)
{
            asm("wfi");
}
+ Include in every SoC/Make.defs(Ex: bcm4390x/Make.defs)